### ISSUE TITLE
Using own dataset for time-series forecasting

### ISSUE THREAD

**Issue by joyjitchatterjee** [maxjcohen/transformer#7]
Hi,
Firstly, thanks for such an interesting repo. I am fairly new to using transformers for Time Series Forecasting, so apologies if my questions are basic. After having gone through the Docs, I have a few queries:-

> 1. How can we use the Transformer model here for forecasting on our own datasets? Suppose, I have a Pandas dataframe with 20000 rows and 34 columns, wherein each column has a measurement parameter.  For example:-

```
DateTime       Parameter1    Parameter2........... Parameter34
2020-01-01.     1.2.                   2.3                          6.3
2020-01-02.    4.4.                  1.4                           1.5
2020-01-03.    1.3                    5.6                           5.5
etc.
```
For now, it seems I can pass my own dataframe directly to dataLoader Class instead of the ozeDataset. But I am not sure if this is the correct way?

> 2. From the docs, I understand that d_input and d_output are related to the dimensions of the dataset. I believe if we are predicting 34 parameters, the d_output would be 34, but how is d_input decided please?

> 3.  For making predictions using 
> predictions = np.empty(shape=(len(dataloader.dataset),k1 ,k2))
> Is k1 and k2 related to the intervals of prediction windows? For example, if we want to predict 34 parameters, how to decide k1 and k2.

Thanks a lot. I am sure this Repo will be very useful for many others, but would be great if you could address these basic questions.

**Comment by maxjcohen** [comment]
Hi Joyjit,
1. The dataset class I have created is very specific to the Oze prediction problem. If you want to use the Transformer model for your own dataset, I encourage you to implement a new dataset class, inheriting from `torch.utils.data.Dataset`, as [explained in the documentation](https://pytorch.org/tutorials/beginner/data_loading_tutorial.html).
2. As we are discussing in #5 , the Transformer is a coherent many to many algorithm, meaning that it takes as input one vector to predict one output vector. For example in the Oze dataset, the input vector is the weather (with dimension `d_input`), and the output is the indoor temperature (with dimension `d_output`).
3.
```python
predictions = np.empty(shape=(len(dataloader.dataset),k1 ,k2))
```
These `k1` and `k2` values should not be hard coded, this is a shortcoming a my behalf. Here we define `predictions` as an array to store the prediction for the entire testing dataset, `k1` is the time dimension (here 168 as the number of hours in a week) and `k2` is the output dimension, here `d_output=8`.

**Comment by joyjitchatterjee** [comment]
Thank you! Much appreciated. I see, yes, I now understood that we can create a new dataset class to load our own datasets, and would try this soon. One more thing please is that the outside usage (out of box) link in the Readme seems to be broken https://timeseriestransformer.readthedocs.io/en/latest/Transformer.html .

**Comment by maxjcohen** [comment]
Thanks, fixed in 4ebc9f5b97ada1689871df8e47e836d307472479. I'm closing this issue for now, don't hesitate to open a new one if you have more questions.
